src/DataStructures/Tensor/EagerMath/Magnitude.hpp

```diff
@@ -28,6 +30,15 @@ Scalar<DataType> magnitude(
   return Scalar<DataType>{sqrt(get(dot_product(vector, vector)))};
 }
 
+template <typename DataType, typename Index>
+void magnitude(
+    const gsl::not_null<Scalar<DataType>*> magnitude,
+    const Tensor<DataType, Symmetry<1>, index_list<Index>>& vector) noexcept {
+  dot_product(magnitude, vector, vector);
```

Comment:
I think these should probably have a `destructive_resize` call in them, no? This applies to pretty much all functions in this PR, I think.